packages/utils/interactive-hooks/src/__tests__/useKeyProps.test.tsx

```diff
@@ -0,0 +1,31 @@
+import * as React from 'react';
+import * as renderer from 'react-test-renderer';
+import { Pressable } from 'react-native';
+import { useKeyProps } from '../useKeyProps';
+import { checkRenderConsistency, checkReRender } from '@fluentui-react-native/test-tools';
+import { PressablePropsExtended } from '../usePressableState.types';
+
+const dummyFunction = () => {
+  console.log('dummy');
+};
+
+// Simple wrapper function to let us use `PressablePropsExtended` to fix type errors
+const PressableWithDesktopProps = (props: PressablePropsExtended) => {
+  return <Pressable {...props} />;
+};
+
+it('Pressable with useKeyProps', () => {
+  const keyboardProps = useKeyProps(dummyFunction, ' ', 'Enter');
+  const tree = renderer.create(<PressableWithDesktopProps {...keyboardProps} />).toJSON();
+  expect(tree).toMatchSnapshot();
+});
+
+it('Simple Pressable with useKeyProps rendering does not invalidate styling', () => {
+  const keyboardProps = useKeyProps(dummyFunction, ' ', 'Enter');
+  checkRenderConsistency(() => <PressableWithDesktopProps {...keyboardProps} />, 2);
+});
+
+it('Pressable with useKeyProps re-renders correctly', () => {
+  const keyboardProps = useKeyProps(dummyFunction, ' ', 'Enter');
+  checkReRender(() => <PressableWithDesktopProps {...keyboardProps} />, 2);
+});
```

Comment:
These tests pass... so I think https://github.com/microsoft/fluentui-react-native/pull/2364 is not actually blocked by more Pressable / useKeyProps changes. 

However, I did speak with @rurikoaraki and an alternate way to land that PR is to remove unnecessary uses of `useKeyProps` from some of our components. On Windows/win32, Pressable by default calls `onPress` with Space/Enter. That's the majority of our use case in FURN, so I can just remove them. I need to implement that default handling in macOS too though, which I'm doing here: https://github.com/microsoft/react-native-macos/pull/1614

Reply:
This is passing in the result of useKeyProps and then using the same result twice to render. This isn't calling useKeyProps twice, which is what's causing the issue in #2364.

Reply:
I think the test that would help is a test that calls useKeyProps twice with the same inputs and verifies that the output is the same object.

Reply:
I added that test. I'm somewhat confused why I can run a hook outside of a function component, but maybe jest tests are special?